File: tree/Traversal/Postorder/recursive/python/PostorderTraversal.py
from collections.abc import Sequence


def PostorderTraversal(left_children: Sequence[int], right_children: Sequence[int], root: int) -> list[int]:
    order: list[int] = []

    def traverse(node: int) -> None:
        left_child = left_children[node]
        if left_child != -1:
            traverse(left_child)

        right_child = right_children[node]
        if right_child != -1:
            traverse(right_child)

        order.append(node)

    traverse(root)
    return order

File: tree/Traversal/Postorder/recursive/rust/src/PostorderTraversal.rs
#[allow(non_snake_case)]
pub fn PostorderTraversal(left_children: &[i32], right_children: &[i32], root: usize) -> Vec<usize> {
    let mut order = Vec::new();

    fn traverse(node: usize, left_children: &[i32], right_children: &[i32], order: &mut Vec<usize>) {
        let left_child = left_children[node];
        if left_child != -1 {
            traverse(left_child as usize, left_children, right_children, order);
        }

        let right_child = right_children[node];
        if right_child != -1 {
            traverse(right_child as usize, left_children, right_children, order);
        }

        order.push(node);
    }

    traverse(root, left_children, right_children, &mut order);
    order
}
